File: Vector2D/C#/Vector2D.cs
using System;

namespace emsp
{
	public class Vector2D
	{
		double x, y;
		Vector2D(double x, double y) {
			this.x = x;
			this.y = y;
		}
		Vector2D(double length, double a, bool T)
		{
			this.y = System.Math.Sin(a) * length;
			this.x = System.Math.Cos(a) * length;
		}
		Vector2D(Vector2D v) {
			this.x = v.x;
			this.y = v.y;
		}
		Vector2D() {
			this.x = 0;
			this.y = 0;
		}
		void SetLenght(double lenght)
		{
			double tempL = Size() / lenght;
			this.x *= tempL;
			this.y *= tempL;

		}
		void SetA(double a)
		{
			this.y = System.Math.Sin(a) * Size();
			this.x = System.Math.Cos(a) * Size();
		}
		double GetA()
		{
			return System.Math.Asin(this.y / Size());
		}
		double Size()
		{
			return System.Math.Sqrt(System.Math.Pow(x, 2) + System.Math.Pow(y, 2));
		}
		double MylScalar(Vector2D v)
		{
			return v.x * v.y + this.x * this.y;
		}

		void MylScalar(int a)
		{
			this.x *= a;
			this.y *= a;
		}
		void Add(Vector2D v)
		{
			this.x += v.x;
			this.y += v.y;
		}

		void Sub(Vector2D v)
		{
			this.x -= v.x;
			this.y -= v.y;
		}
		~Vector2D()
		{
		}
	}
}

File: Vector2D/Vector2D.cpp
#include "Vector.h"
namespace emsp {
	Vector::Vector(float x, float y)
	{
		this->x = x;
		this->y = y;
	}
	Vector::Vector(float length, float a, bool T)
	{
		this->y = sin(a) * length;
		this->x = cos(a) * length;
	}
	Vector::Vector(Vector& v) {
		this->x = v.x;
		this->y = v.y;
	}
	Vector::Vector() {
		this->x = 0;
		this->y = 0;
	}
	void Vector::setLenght(float lenght)
	{
		double tempL = size() / lenght;
		this->x *= tempL;
		this->y *= tempL;

	}
	void Vector::setA(float a)
	{
		this->y = sin(a) * size();
		this->x = cos(a) * size();
	}
	float Vector::getA()
	{
		return asin(this->y / size());
	}
	float Vector::size()
	{
		return sqrt(pow(x, 2) + pow(y, 2));
	}
	float Vector::mylScalar(Vector v)
	{
		return v.x * v.y + this->x * this->y;
	}

	void Vector::mylScalar(int a)
	{
		this->x *= a;
		this->y *= a;
	}
	void Vector::add(Vector v)
	{
		this->x += v.x;
		this->y += v.y;
	}

	void Vector::sub(Vector v)
	{
		this->x -= v.x;
		this->y -= v.y;
	}
	Vector::~Vector()
	{
	}
}
